Plasma Beaker

Kurfursten opened the pull request:
Someone in the thread suggested this, I thought it was a cute idea

This beaker acts as a catalyst for the three plasma reactions (Dexalin, Clonexadone, Leporazine) so you don't need any plasma reagent inside. It holds up to 100 units. It requires the highest plasmatech you can reach, 4, as well as 4 materials. It takes ~8 sheets of plasma to make, or less than 5 with upgrades.

New Sprite below

Fully tested

Commandersand commented:
SMASH THAT MF OXYCODONE BUTTON

Kurfursten commented:
Oxycodone doesn't use plasma as a catalyst, it uses it as a component.

Commandersand commented:
heck

iFlashYou commented:
18000 not a bit much? How about only 8000?

Kurfursten commented:
As anyone who plays in science can tell you, plasma enjoys the benefits of always being in surplus. It pays out as much as diamonds to miners but it's the most common ore type. On top of that, it's used in almost no designs. In practice, the difference between 18000 and 8000 is whether a miner mines 1 tile of plasma or 2 tiles of plasma.

I'm not too stubborn on that point, it's just that I figured it would be nice if science had a way to get rid of the mountains of that stuff

iFlashYou commented:
Tbh, its less about the spent materials and rather about the time it takes to make one.

Kurfursten commented:
It didn't take very long in testing, I actually stood there and did the plasma + materials tech path and built it from the protolathe

N3X15 commented:
* Added [balance] because affects ease of chemistry manufacture

Cyadox commented:
As someone who does chemistry often, I love this idea. Typically I avoid plasma catalyst chems like the plague after I get my Dex+/Cryo/Clonex done, just because you have to fuck around a lot not to lose the very little plasma you have. Also making it the size of a large beaker means I'd still want bluespace/stasis beakers, which is a nice touch.

Now that being said, I can see some issues. Beyond roundstart cryomixes, its usefulness is drastically diminished. There's rarely a reason to mass produce Dex+/Cryo/Clonex, although Leporazine could see a lot more ab/use, and it's still only a large beaker. I would happy to see some downsides to using it as well - maybe it could be volatile, or have a chance to exude plasma gas if dropped/reacted with too much.

Commandersand commented:
Make it poison anyone who drinks from it

nuklearcellphoneg commented:
I'd pop out plenty of Lepo whenever I'd do clonex- of course, I never bothered with dplus plasma reactions since I'd just drop a dex pill into carbon-iron and pop out 90u of dplus.

So, lexorin and oxycodone aren't a part of this at all?

Cyadox commented:
@nuklearcellphoneg
It's hardcoded to only work on Cryox, Dex+ and Lepo

Kurfursten commented:
More or less my thinking. As for drawbacks, I didn't really see the need, much like there's no "drawback" for a stasis beaker or a bluespace beaker, it's just gated by research and resources.

I'm pretty easygoing about this little PR though, so if you can come to any consensus on changes I'll do whatever

nuklearcellphoneg commented:
Like has been pointed out, kinda meh. Neat, but the requirements for it are stronk and it's largely just taking the place of 5u of plasma. Lepo and Clonex are both early game things, sabotage does happen sometimes but now it can be extended to 'dump out plasma, cryo, and space the plasma beaker', and dplus doesn't need plasma since you need only access one of the 20 oxy dep kits around the station.

Kind of neat, price is way steep, usefulness is a bit limited. Prolly more meaningful for ghetto chem, and not having any use in regards to oxy and lex is a bit of a shame.

Unknown-as-Captain commented:
It's just diluting the pool of shit Science can print so they bring you the stuff you actually want even less often. Plasmemesaw did the same thing.

Blithering commented:
![plasmabeaker](https://user-images.githubusercontent.com/15067109/30936216-07e2a16c-a3cb-11e7-936d-c91aa120cafe.png)
beaker with suspended plasma?

Cyadox commented:
I'll have to agree with nuklear. It's a nice idea but I don't see when it'd be used. 
Chemistry is more likely to have all those chems done with their roundstart plasma than science is going to have all this research done, have materials delivered, AND make it.

Chucklemonger commented:
The only problem I see is that if the Medbay staff gives enough of a shit to make a cryo mix and such, they'll probably have it done before this becomes available. I don't particularly see a reason to not merge it, though. It's a neat idea and it might see niche use.

Kurfursten commented:
I suppose I could toss it in the CMO's locker instead and just make it a perk item, like the hypospray/holoswitchtool/advanced toolbelt/lawgiver

SkowronX commented:
bOI

Commandersand commented:
Put it on the CMO's office table if you can so that medbay can just ask the AI if there's no CMO.
It's not really high risk and people can finish the cryomix by the time you get the AI to open it.

Blithering commented:
Maybe make it available via nanomed voucher?

nuklearcellphoneg commented:
I'd like to clarify it's neat fluff.

Putting it into the CMO's office would necessitate an extra step in sabotaging medbay and cryo that I think would be too much.

I'm not saying it's bad as is, just very niche like CPR.

Commandersand commented:
That'd be cool, I'd like to be able to make cryo mix as a gray shirt without bothering science for mats
 Maybe put the voucher in a medbay crate?
Does the beaker now make clonexadone? That would be good too

Commandersand commented:
>extra step to sabotaging medbay and cryo

Doesn't that just require pressing one button on the oxygen tank?

nuklearcellphoneg commented:
If you dump out cryo, dump out clonex and cryox, dump out plasma, you've either got to access Science or Mining one. Messing with the air sends you to first the Paramed's office, second to one of the uh... emergency rooms? You know, the ones with coolers, heaters, an air tank, etc. I'd argue that when you don't have it, Plasma can be pretty hard to get without help, especially for a doc's access and loadout.

Kurfursten commented:
It makes the three things that have plasma as a catalyst, which yes, includes clonex

Mapping it in is slightly more annoying because it requires mapping

nuklearcellphoneg commented:
Yes pls don't map it in thx

This is probably underappreciated as well, but being a newbie chemist, the plasma catalyst teaches you about syringes, and 5u, and beaker pour amounts, and splashing, removing x units when making pills, and so on, in ways that most veterans likely take for granted.

Commandersand commented:
or you can just dismantle cloning to sabotage medbay
Only bad doctors put non-clones in cryo anyways :^)

nuklearcellphoneg commented:
The difficulty in dismantling/destroying cloning is much greater than plasma, as well as the fact that getting rid of plasma and its derivatives is very subtle, whereas cloning being destroyed/dismantled sticks out like a sore thumb.

Commandersand commented:
Really?
I've only seen rods and a non antag captain destroy cloning but I'm sure you can screwdriver+crowbar it and take the board to cause panic

nuklearcellphoneg commented:
And then it becomes wildly different.

If you sabotage by getting rid of clonex, plasma, so on, people won't realize it's sabotaged until they must to clone. Okay, there's no meat, we need meat. Then they go to chem; there's no clonex. So they go get a monkey, butcher it, put in meat. Now we can clone this guy, and they do so. Now they drag him to cryo, put him in- he's not healing, why? There's no cryox or clonex. Okay, so let's go make some. Oh, there's no plasma, and so on. 

If you sabotage the cloner itself, they see it's sabotaged and may preemptively fix it.

Exxion commented:
1. Have someone IED cryo so that the floor and piping is damaged but not destroyed
2. While repairing it, secretly connect the cryo pipes to waste
3. Watch medbay panic over cryo not working and probably never figure out why

nuklearcellphoneg commented:
I'd compare it to the robustness of storing traitor items in a trash bin by turning off the pump. Someone might notice that little light, but more than likely, no one will notice until they are looking for something or someone that's hidden, not before.

Commandersand commented:
I GUESS

nuklearcellphoneg commented:
@Exxion 

1. Make a complicated mix with 5u of every healing chem
2. Kill your medbay cohorts
3. There's also 5u of leporazine in the mix
4. Kill anyone who knows/figures out what leporazine does

Kurfursten commented:
Clearly there's a lot of discussing going on but it's hard to read any kind of concensus

👍 for fine as-is (built from protolathe)
Hooray for "give it to the CMO"
❤️  for "voucher from cargo" instead

Zth-- commented:
tl;dr, maybe someone suggested it before, but add it to the trader's vending machine! That way you can have early shift tradings between them and medbay

CptWad commented:
you could have it require NT research level instead

Shadowmech88 commented:
@Kurfursten 
I feel like it might have been better to use :-1: for the second option since you can put it in your post.
Presently, at a glance it might seem to some people as though there are only two options.

Kurfursten commented:
I have made the following changes

* Lowered the cost to 12000 plasma, which is a decrease of 33%
* Updated sprite (chemical.dmi) and added a new reagentfilling sprite series for the plasma beaker. Due to this sprite's size, it was necessary to create discrete fillings.
* Resolved conflicts

I tested it again after these changes to ensure the reagent filling matched up correctly with the new sprite

Hugh-G-Rection commented:
Now make a universal enzyme drinking glass.

Intigracy commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -910,6 +910,31 @@
 
+// Special Reactions for Plasma Beaker
+/datum/chemical_reaction/plasmabeakerdexalin
+	name = "Plasma Beaker Dexalin"
```
I feel like this is messier than just making them subtypes of the recipes (right under the relevant recipes) and overriding the two vars (the required chems and required container) as opposed to copypasting all of them

Kurfursten replied:
That seems valid but it's consistent with how it's done everywhere else. e.g.: just above here, you can see the botany reactions.
